File: src/DotVast.HashTool.NativeCrypto/RIPEMD.cs
// Copyright (c) Kiyan Yang.
// Licensed under the MIT License.

namespace DotVast.HashTool.NativeCrypto;

[NativeCrypto("ripemd128", 16)]
public partial class RIPEMD128 : NativeCryptoBase
{
}

[NativeCrypto("ripemd160", 20)]
public partial class RIPEMD160 : NativeCryptoBase
{
}

[NativeCrypto("ripemd256", 32)]
public partial class RIPEMD256 : NativeCryptoBase
{
}

[NativeCrypto("ripemd320", 40)]
public partial class RIPEMD320 : NativeCryptoBase
{
}

File: src/DotVast.HashTool.NativeCrypto/native_crypto/src/ripemd.rs
use crate::hasher::impl_hasher;
use ripemd::{Ripemd128, Ripemd160, Ripemd256, Ripemd320};

impl_hasher!("ripemd128", Ripemd128);
impl_hasher!("ripemd160", Ripemd160);
impl_hasher!("ripemd256", Ripemd256);
impl_hasher!("ripemd320", Ripemd320);
